README.md

```diff
@@ -43,3 +43,11 @@ Then you can query the server like this:
 ``` bash
 curl localhost:3000/item
 ```
+
+### Nix
+
+Using Nix or NixOS, you can run `nix-shell stack.nix` before
+following the above instructions. But please note, you'll want
+to edit `stack.nix` and comment out large visual studio code
+dependencies if you don't want to pull these in 
+(`hiepkgs.hies`, `vscode`).
```

Comment:
I'm not really happy about the idea of having two different ways to build this package that we need to maintain. And since I don't have `nix` installed on my system, I'm against switching to a build process that requires `nix`. Does this not build for you with just `stack` (with the old resolver)? What exact problem are you trying to solve?

I think one issue that we should fix is to switch to newer versions of servant and other dependencies. But that should definitely be done in a separate PR without switching the build system.

Reply:
Oh, I wasn't switching the build system. This is just a way to fire up a development environment (or build environment) in nix to get the ball rolling. As listed in the nix file, a few things are needed, which the nix expression provides. At a minimum, cabal-install, stack, and gmp seemed to be needed for me.

I'll admit I don't think the related change to stack.yaml requires nix, just that it will use it if you have it, though I should verify that.

But sure, I can do this in a separate PR.

Reply:
Or, if you are able to test this PR on your system, you can let me know if the stack.yaml change breaks anything for you.

Reply:
Yeah, that's what I meant by build system. :) I am currently using `stack` to spin up a development environment, and I don't really want to have two ways of doing that.

I think ideally the README in this repo would have one easy recommended and reproducible way of spinning up a development environment. And hopefully that recommended procedure works everywhere. Currently the README says to use `stack setup` and `stack test`, etc.. Does that not work for you?

Your changes _do_ break things for people that don't have `nix`. This repo has travis CI setup, there you can see the error: https://travis-ci.org/haskell-servant/example-servant-minimal/builds/470227505?utm_source=github_status&utm_medium=notification.

Reply:
Thanks for pointing that out. I fixed the issue by removing the `enable: true` under the nix section, so it no longer forces nix to be used.

At the risk of getting off the main point of the PR (updating to work on recent LTS), the point I'm trying to make is that Nix is not a separate build system, and should not get in the way for any non-Nix user. The Nix stuff here is pulling in system dependencies needed to do the build (cabal, gmp -- not a haskell interface to gmp, and stack). Without these, on nix, you cannot build it; stack still does the actual build. This is just a convenient way to allow Nix users to get an environment set up quickly. In Nix, you tend to not have all your software available globally - the standard practice is to have individualized environments available for each project. Much like a container, but not a container, and not as heavy-weight as a container, but now I'm really getting off topic. A Nix user shouldn't even have stack and nix in their global enviornment to begin with. Not sure why gmp is needed or not pulled in by other deps.

Reply:
I would like to limit the scope of this PR to updating to a new stack snapshot. Would you be up for removing all the nix related things?

We can open another issue or PR where we then can discuss whether we want to add anything nix related to the stack file.